C/include/c4ConnectedClient.h

```diff
@@ -37,15 +68,47 @@ C4ConnectedClient* c4client_new(const C4ConnectedClientParameters* params,
  @param asFleece  If true, the response's `body` field is Fleece; if false, it's JSON.
  @param callback Callback for getting document.
  @param context Client value passed to getDocument callback
- @param error  On failure, the error info will be stored here. */
+ @param outError  On failure, the error info will be stored here. */
 void c4client_getDoc(C4ConnectedClient*,
                      C4Slice docID,
                      C4Slice collectionID,
                      C4Slice unlessRevID,
                      bool asFleece,
                      C4ConnectedClientGetDocumentCallback callback,
                      void * C4NULLABLE context,
-                     C4Error* error) C4API;
+                     C4Error* outError) C4API;
+
+/** Callback for updating the document result.
+ 
+ @param client   The client that initiated the callback.
+ @param err Error will be written here if the get-document fails.
+ @param context  user-defined parameter given when registering the callback. */
+typedef void (*C4ConnectedClientUpdateDocumentCallback)(C4ConnectedClient* client,
+                                                        C4HeapSlice revID,
+                                                        C4Error* C4NULLABLE err,
+                                                        void * C4NULLABLE context);
+
+typedef C4ConnectedClientUpdateDocumentCallback C4ConnectedClientUpdateDocumentCallback;
+
```

Comment:
This should go in c4ConnectedClientTypes.h.

Also, the second typedef is unnecessary AFAIK.